### PR TITLE
Connect metric hub glossary terms to source tables

### LINKED ISSUE #93: Attach glossary terms to datasets via yaml
See [this ticket](https://acrylsupport.zendesk.com/hc/en-us/requests/1075) and [this ticket](https://mozilla-hub.atlassian.net/browse/DENG-3296) for more context.

### PR DESCRIPTION
**Pull request by Iinh**
Fixes #93

### CODE REVIEW

**Comment by Iinh** [comment]
@ANich could you give an initial review? thanks!

File: sync/datahub/metrichub_glossary.py [hunk 1]
```diff
@@ -90,6 +93,20 @@ def _build_product_dict(product: str, metrics: List[MetricHubDefinition]) -> Dic
     }
 
 
+def _connect_terms_to_source_table(metrics: List[MetricHubDefinition]) -> None:
```
Comment (ANich) [inline sync/datahub/metrichub_glossary.py:96]: Not `None` if it's gonna return the dictionary. You could also just write the yaml directly in this function too
```suggestion
def _connect_terms_to_source_table(metrics: List[MetricHubDefinition]) -> Dict[...]:
```

**Comment by Iinh** [comment]
@ANich could you do a final pass for this PR? I opened a [separate PR](https://github.com/mozilla/mozilla-datahub-ingestion/pull/110) for the tests because they cover more than the changes here.

**Review by ANich — APPROVED** [review]
LGTM if you've tested

**Pull request merged**